riscv-sbi.adoc

```diff
@@ -1674,6 +1675,250 @@ The possible error codes returned in `sbiret.error` are shown in the
+https://trustedcomputinggroup.org/wp-content/uploads/TCG_D-RTM_Architecture_v1-0_Published_06172013.pdf[TCG specified DTRM PCR indexes].
+
+All measurements added through this SBI function must be included in the TCB
+attestation evidence generated by the <<get-evidence-function>> function.
+
+The measurement at the `measurement_addr` physical address must be generated
+by a hash function that uses the algorithm specified by `hash_algorithm`.
+`hash_algorithm` must be one of the supported hash algorithms, as reported by
+the `HASH_ALGORITHMS` field from the <<get-capabilities-function>> returned
+structure (`ATST_CAPS`). See
+<<table_attestation_and_measurement_capabilities_format>> for more details.
+
+The possible error codes returned in `sbiret.error` are shown in the
+<<table_measurement_extend_errors>> below.
+
+[#table_measurement_extend_errors]
+.Measurement Extend Errors
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code            | Description
+| SBI_SUCCESS           | The measurement was succesfully extended.
+| SBI_ERR_FAILED        | The measurement could not be extended.
+| SBI_INVALID_PARAM     | The `measurement_index` is invalid.
+| SBI_ERR_NOT_SUPPORTED | The function is not supported as the measurement
+                          can not be extended.
+|===
+
+[#get-capabilities-function,reftext="Get Capabilities"]
+=== Function: Get Attestation and Measurement Capabilities (FID #2)
+
+[source, C]
+----
+struct sbiret sbi_attestation_get_capabilities(unsigned long caps_addr)
+----
+
+Get the SBI implementation attestation and measurement capabilities.
+
+The reported capabilities allow attestation and measurement implementations to
+build platform agnostic frameworks, by dynamically adapting the SBI calls to the
+platform capabilities.
+
+The SBI implementation places an attestation and measurement capabilities
+structure at the caller's `caps_addr` physical address. The capabilities
+structure format follows the ÀTST_CAPS`format describe in
+<<table_attestation_and_measurement_capabilities_format>>.
+
+It is the caller's responsibility to allocate enough memory at `caps_addr` to
+accomodate for the largest possible capabilities structure, which is 136 bytes.
+The actual capabilities structure length is returned through `sbiret.value`.
+
+[#table_attestation_and_measurement_capabilities_format]
+.Attestation and Measurement Capabilities Format Definition (ATST_CAPS)
+[cols="5,1,1,1,5", width=80%, align="center", options="header"]
+|===
+| Field Name            | Offset (bytes) | Type           | Size (bytes)           | Description
+| TCB_SVN               | 0              | Integer        | 4                      | TCB Security Version Number
+| HASH_ALGORITHMS       | 4              | Bitmap         | 1                     a| Supported measurement hash algorithms bitmap:
+
+                                                                                     * Bit 0: SHA-384
```

Comment:
Out of curiosity, is SHA-256 no longer being used for this purpose? The reason for asking is that TPM2 is still SHA256.

Reply:
SHA 256 is not secure against length extension attacks, and none of the existing TEE implementations (TDX, SEV, SE, etc) use it. They use SHA-384.